src/handlers/discord/voiceStateHandler.js

```diff
@@ -0,0 +1,583 @@
+    this.DG.voiceSettings.prevVoiceActivityData = newData;
+    const states = [];
+    const connectors = [];
+
+    if (data.hasOwnProperty('input')) {
+      // updating the input devices and names as they may have changed
+
+      // Making sure the data is valid..
+      if (Array.isArray(data.input.available_devices) && data.input.available_devices.some(device => device.id && device.name)) {
+        // Process the input devices
+        this.DG.voiceSettings.inputDevices = data.input.available_devices;
+        this.DG.voiceSettings.inputDeviceNames = Object.keys(this.DG.voiceSettings.inputDevices).map(key => this.DG.voiceSettings.inputDevices[key].name);
+
+        this.DG.voiceSettings.inputDeviceVolume = data.input.volume;
+        this.DG.voiceSettings.inputDeviceId = data.input.device_id;
+        
+        // Matching the Default Device to Name
+        let matchedDevice = this.DG.voiceSettings.inputDevices.find(device => device.id === this.DG.voiceSettings.inputDeviceId);
+        // this.TPClient.stateUpdate("discord_inputDevice", String(matchedDevice.name));
+        states.push({id: "discord_inputDevice", value: String(matchedDevice.name)});
+        logIt("DEBUG", "Input Device: ", String(matchedDevice.name));
+      }
+    } else {
+      logIt("INFO", "No valid input devices found.");
+    }
+    
+
+    if (data.hasOwnProperty('output')) {
+      if (Array.isArray(data.output.available_devices) && data.output.available_devices.some(device => device.id && device.name)) {
+
+        this.DG.voiceSettings.outputDevices = data.output.available_devices;
+        this.DG.voiceSettings.outputDeviceNames = Object.keys(this.DG.voiceSettings.outputDevices).map(key => this.DG.voiceSettings.outputDevices[key].name);
+
+        this.DG.voiceSettings.outputDeviceVolume = data.output.volume;
+        this.DG.voiceSettings.outputDeviceId = data.output.device_id;
+
+        // Matching the Default Device to Name
+        let matchedDevice = this.DG.voiceSettings.outputDevices.find(device => device.id === this.DG.voiceSettings.outputDeviceId);
+        states.push({id: "discord_outputDevice", value: String(matchedDevice.name)});
+        logIt("DEBUG", "Output Device: ", String(matchedDevice.name));
+      }
+      } else {
+        logIt("INFO", "No valid input devices found.");
+      }
+
+    if (data.hasOwnProperty("mute")) {
+      if (data.mute) {
+        this.DG.voiceSettings.muteState = 1;
+      } else {
+        this.DG.voiceSettings.muteState = 0;
+      }
+
+      logIt("DEBUG", `discord mute is ${this.DG.voiceSettings.muteState} `);
+      states.push({id: "discord_mute", value: this.DG.voiceSettings.muteState ? "On" : "Off"});
+    }
+    if (data.hasOwnProperty("deaf")) {
+      if (data.deaf) {
+        this.DG.voiceSettings.deafState = 1;
+        // this.DG.voiceSettings.muteState = 1; // deaf is deaf.. why would we turn off mute? i understand if they are deaf they are mute also but thats up to the user to decide??
```

Comment:
Lets resolve the issue with mute state not being effected when a user deafens

```suggestion
        this.DG.voiceSettings.muteState = 1;
```